go/plugin/openai-a2a/router.go

```diff
@@ -0,0 +1,343 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/agentic-layer/agent-gateway-krakend/lib/logging"
+	"github.com/agentic-layer/agent-gateway-krakend/lib/models"
+	"github.com/go-http-utils/headers"
+)
+
+// responseWriter wraps http.ResponseWriter to capture response body
+type responseWriter struct {
+	http.ResponseWriter
+	body       *bytes.Buffer
+	statusCode int
+}
+
+func newResponseWriter(w http.ResponseWriter) *responseWriter {
+	return &responseWriter{
+		ResponseWriter: w,
+		body:           &bytes.Buffer{},
+		statusCode:     http.StatusOK,
+	}
+}
+
+func (rw *responseWriter) Write(b []byte) (int, error) {
+	return rw.body.Write(b)
+}
+
+func (rw *responseWriter) WriteHeader(statusCode int) {
+	rw.statusCode = statusCode
+}
+
+// ModelInfo contains information about an agent model
+type ModelInfo struct {
+	Name      string
+	Namespace string
+	URL       string
+}
+
+// parseModelParameter parses and validates the model parameter.
+// Requires format: "namespace/agent-name"
+// Returns an error if the model parameter doesn't conform to the expected format.
+func parseModelParameter(model string) (namespace string, agentName string, err error) {
+	if model == "" {
+		return "", "", fmt.Errorf("model parameter cannot be empty")
+	}
+
+	// Check for path traversal attempts
+	if strings.Contains(model, "..") {
+		return "", "", fmt.Errorf("invalid model parameter format")
+	}
+
+	// Requires format: namespace/agent-name
```

Comment:
Do we actually have to validate this format here? Why not leave the format to the operator.